src/platform_impl/linux/x11/util/randr.rs

```diff
@@ -94,38 +101,49 @@ impl XConnection {
         );
         let name = String::from_utf8_lossy(name_slice).into();
         // Override DPI if `WINIT_HIDPI_FACTOR` variable is set
-        let dpi_override = env::var("WINIT_HIDPI_FACTOR")
-            .ok()
-            .and_then(|var| f64::from_str(&var).ok());
-        let hidpi_factor = if let Some(dpi_override) = dpi_override {
-            // `0` makes winit use its auto dpi calculation mechanism bypassing Xft.dpi
-            if dpi_override == 0. {
-                calc_dpi_factor(
-                    ((*crtc).width as u32, (*crtc).height as u32),
-                    (
-                        (*output_info).mm_width as u64,
-                        (*output_info).mm_height as u64,
-                    ),
-                )
-            } else {
+        let dpi_env = env::var("WINIT_HIDPI_FACTOR").ok().map_or_else(
+            || EnvVarDPI::NotSet,
+            |var| {
+                if var.to_lowercase() == "randr" {
+                    EnvVarDPI::Randr
+                } else if let Ok(dpi) = f64::from_str(&var) {
+                    EnvVarDPI::Scale(dpi)
+                } else {
+                    EnvVarDPI::NotSet
```

Comment:
It looks like this is treating an invalid env var value the same as an absent env var. Is that the desired behavior?

Reply:
We allowed arbitrary strings, so the idea was to follow the old behavior and not to blow up here.  If you feel like we should blowup let me know, however I can see an issue if we add one more env var and user will have 2 winit apps with different winit version.